### PR TITLE
[WAIT for testing in stage] Use background job-based endpoint for virtual object creation

### PR DESCRIPTION
**Pull request by mjgiarlo**
Connects to #1323 
Connects to sul-dlss/dor-services-app#336

~Blocked by https://github.com/sul-dlss/operations-tasks/issues/2041~
~Blocked by https://github.com/sul-dlss/dor-services-app/issues/447~

## Why was this change made?

This PR allows Argo to use the new virtual objects endpoint in dor-services-app that processes batches of virtual objects in the background. This allows for larger batches of virtual objects to be created.

Includes:
* Bump dor-services-client version to 3.0.0 to be able to use the new endpoint
* Refactor the super awful `CreateVirtualObjectsJob`: I extracted two small objects from the job which greatly reduces its responsibility and knowledge.
* Stop passing around a hash of virtual objects when only an array is needed.
* Do not bother spending rubocop time scanning directories with irrelevant files: this speeds up rubocop execution noticeably.

**Note** that we **CANNOT** merge and deploy this until sul-dlss/puppet#4470 is merged and puppet has run on the affected boxes. cc: @tallenaz

## Was the documentation updated?

No, I don't believe it needs to be?

### CODE REVIEW

**Comment by mjgiarlo** [comment]
Blocked by https://github.com/sul-dlss/dor-services-app/issues/447

**Comment by jcoyne** [comment]
@mjgiarlo needs a rebase too.

File: app/jobs/create_virtual_objects_job.rb [hunk 1]
```diff
@@ -22,85 +20,50 @@ def perform(bulk_action_id, params)
...
+      if problematic_druids.any?
+        log.puts("#{Time.current} #{UNAUTHORIZED_MESSAGE}: #{unauthorized_druids.to_sentence}") if unauthorized_druids.any?
+        log.puts("#{Time.current} #{NOT_FOUND_MESSAGE}: #{not_found_druids.to_sentence}") if not_found_druids.any?
+        bulk_action.increment!(:druid_count_fail, problematic_druids.length)
+
+        # Short-circuit if all parent druids were problematic
+        if problematic_druids.length == virtual_objects.length
+          log.puts("#{Time.current} No virtual objects could be created. See other log entries for more detail.")
+          log.puts("#{Time.current} Finished #{self.class} for BulkAction #{bulk_action_id}")
+          return
```
Comment (mejackreed) [inline app/jobs/create_virtual_objects_job.rb]: What the heck???

Comment (peetucket) [reply app/jobs/create_virtual_objects_job.rb]: 100% heck.

**Comment by mjgiarlo** [comment]
Blocked by a related error in the publish workflow: https://app.honeybadger.io/projects/52894/faults/54832514

**Comment by mjgiarlo** [comment]
Working well, even at scale. Lingering issue: multiple versions apparently simultaneously created, or workflow service reporting bad timestamps. Example virtual object: https://argo-stage.stanford.edu/view/druid:zd904tk7751

**Comment by mjgiarlo** [comment]
@andrewjbtw has approved this work in stage. merging.

**Comment by mjgiarlo** [comment]
Note: I am leaving some debugging stuff (HB alerts) in for now while we roll this out to production. Can remove later.

**Pull request merged**